appstore/validator.go

```diff
@@ -120,7 +121,20 @@ func (c *Client) Verify(req IAPRequest, result interface{}) error {
 	}
 	defer resp.Body.Close()
 
-	err = json.NewDecoder(resp.Body).Decode(result)
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	r := &IAPResponse{}
+	err = json.Unmarshal(bodyBytes, r)
+	if err == nil && r.Status == 21007 {
+		c.URL = SandboxURL
+		err = c.Verify(req, result)
```

Comment:
Try to avoid recursion unless the alternative is much more difficult or less efficient. In this case, without recursion you'd be able to reuse the http client and the request body. Then just do another POST right here with the sandbox URL. Then you don't have to set the URL back after, which seems a little hacky.